src/lib/layer-manager.js

```diff
@@ -446,6 +496,86 @@ export default class LayerManager {
     }
     return error;
   }
+
+  _validateEventHandling() {
+    // Check if a deck-level mouse event has been specified
+    // and that at least one of the layers in the deck is pickable
+    if (
+      this.onLayerClick ||
+      this.onLayerHover
+    ) {
+      if (this.layers.length && !this.layers.some(layer => layer.props.pickable)) {
+        log.once(1,
+          'You have supplied a top-level input event handler (e.g. `onLayerClick`), ' +
+          'but none of your layers have set the `pickable` flag.'
+        );
+      }
+    }
+  }
+
+  /**
+   * Route click events to layers.
+   * `pickLayer` will call the `onClick` prop of any picked layer,
+   * and `onLayerClick` is called directly from here
+   * with any picking info generated by `pickLayer`.
+   * @param {Object} event  An object encapsulating an input event,
+   *                        with the following shape:
+   *                        {Object: {x, y}} offsetCenter: center of the event
+   *                        {Object} srcEvent:             native JS Event object
+   */
+  _onClick(event) {
+    const pos = event.offsetCenter;
```

Comment:
I have asked this before:

Is `offsetCenter` a standard field in the Event? My understanding is that it is NOT. if that is true, this code is implicitly coupled to the `EventManager` because it will only work with events generated by that class.

To break this coupling, `position` parameter to this function and call as
```
layerManager._onClick({position: event.offsetPosition, event});
```